packages/react-server/src/ReactFizzServer.js

```diff
@@ -483,6 +484,48 @@ function renderSuspenseBoundary(
   // context switching. We just need to temporarily switch which boundary and which segment
   // we're writing to. If something suspends, it'll spawn new suspended task with that context.
   task.blockedBoundary = newBoundary;
+  if (
+    enableCPUSuspenseFizz &&
+    typeof props.unstable_expectedLoadTime === 'number'
+  ) {
+    // Create task to Render the fallback now
+    task.blockedSegment = boundarySegment;
+    try {
+      renderNode(request, task, fallback);
+      boundarySegment.status = COMPLETED;
+
+      // Create task to render the content
+      const contentTask = createTask(
+        request,
+        content,
+        newBoundary,
+        contentRootSegment,
+        fallbackAbortSet,
+        task.legacyContext,
+        task.context,
+        task.treeContext,
+      );
+      if (__DEV__) {
+        contentTask.componentStack = task.componentStack;
+      }
+      // Render the content in the next tick, since it is large.
+      setImmediate(() => {
+        pingTask(request, contentTask);
+      });
+      popComponentStackInDEV(task);
+    } catch (error) {
+      boundarySegment.status = ERRORED;
+      logRecoverableError(request, error);
+      newBoundary.forceClientRender = true;
+      // We don't need to decrement any task numbers because we didn't spawn any new task.
+      // We don't need to schedule any task because we know the parent has written yet.
+      // We do need to fallthrough to create the fallback though.
+    } finally {
+      task.blockedBoundary = parentBoundary;
+      task.blockedSegment = parentSegment;
+    }
+    return;
```

Comment:
This might be a case where `if ... else` is better than early return since they're kind of equal branches.